src/group_theory/pi_hom.lean

```diff
@@ -0,0 +1,260 @@
+/-
+Copyright (c) 2022 Jocchim Breitner. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joachim Breitner
+-/
+import group_theory.subgroup.basic
+import tactic.group
+import group_theory.general_commutator
+import group_theory.order_of_element
+import data.finset.noncomm_prod
+
+/-!
+# TODO
+
+-/
+
+variables {G : Type*} [group G]
+
+lemma mul_eq_one_of_disjoint
+  {H₁ H₂ : subgroup G} (hdis : disjoint H₁ H₂) {x y : G} (hx : x ∈ H₁) (hy : y ∈ H₂)
+  (heq : x * y = 1) : x = 1 ∧ y = 1 :=
+begin
+  have : y = x⁻¹ := symm (inv_eq_iff_mul_eq_one.mpr heq),
+  subst this,
+  have hy := H₂.inv_mem_iff.mp hy,
+  have : x ∈ H₁ ⊓ H₂, by { simp, cc },
+  rw [hdis.eq_bot, subgroup.mem_bot] at this,
+  subst this,
+  simp
+end
+
+namespace pi_hom
+
+-- We have an family of group
+variables {I : Type*} [fintype I] [decidable_eq I] {H : I → Type*} [∀ i, group (H i)]
+
+-- And morphism ϕ into G
+variables (ϕ : Π (i : I), H i →* G)
+
+-- We assume that the elements of different morphism commute
+-- Since we need this all over the place we wrap it up in `fact`
+variables [hcomm : fact (∀ (i j : I), i ≠ j → ∀ (x : H i) (y : H j), commute (ϕ i x) (ϕ j y)) ]
```

Comment:
Why do you need this? Is it just because you're using `noncomm_prod`? Is there a reason you can't just use `list.prod` and work with `list`s (possibly subtyped by `list.nodup`) instead?

Reply:
No, I’ll need this assumption in any case, else I won’t be able to show that the function from the product into `G` is actually a homomorphism.